client-js/app/elements/labrad-registry.ts

```diff
@@ -26,14 +26,31 @@ export class LabradRegistry extends polymer.Base {
 
   @property({type: String, notify: true, value: null})
   selectType: string;
+    
+  @property({type: String, notify: true, value: ''})
+  filterText: string;
 
   //Helper Functions
   @observe('path')
   pathChanged(newPath: string[], oldPath: string[]) {
-    // on a path change, we deselect everything
+    // on a path change, we deselect everything, empty filterText
     this.selDir = null;
     this.selKey = null;
     this.selectType = null;
+    this.filterText = '';
+  }
+
+  @observe('filterText')
+  reloadMenu() {
+    //triggers re-render of dir, key lists when filterText is changed
+    this.$.dirList.render();
+    this.$.keyList.render();
+  }
+
+  filterFunc(item) {
+    // called when dir, key lists are populated. Returns entries that contain
+    // substring in filterText
+    return item.name.match(new RegExp(this.filterText, 'i'));
```

Comment:
It would be nice to avoid recompiling this regular expression for every item. I would suggest creating it once in the `reloadMenu` method and storing it in a member variable.